src/components/fullscreen-flyout/FullscreenFlyout.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+import * as React from 'react';
+
+import './FullscreenFlyout.scss';
+import classNames from 'classnames';
+import { Flyout, Overlay } from '../flyout';
+import OverlayHeader from './OverlayHeader';
+
+export type FullscreenFlyoutProps = {
+    /** Components to render in the overlay */
+    children: React.Node,
+    /** Set className to the overlay wrapper */
+    className?: string,
+    /** Custom <OverlayHeader> */
+    header?: React.Element<any>,
+    /** CSS selector that identifies additional clickable elements that closes the overlay */
+    onCloseClassSelector?: string,
+};
+
+type Props = FullscreenFlyoutProps;
+
+const FullscreenFlyout = (props: Props) => {
+    const { children, className, header, onCloseClassSelector, ...rest } = props;
+
+    const elements = React.Children.toArray(children);
+
+    if (elements.length !== 2) {
+        throw new Error('FullscreenFlyout must have exactly two children: A button component and a content element');
+    }
+
+    const [button, content] = elements;
+
+    // Close flyout button in header should be clickable by default if using selector override
+    const clickableSelector = onCloseClassSelector ? `.close-btn, ${onCloseClassSelector}` : '.close-btn';
+
+    return (
+        <Flyout
+            disableTether
+            onCloseClick
+            clickableOverrideSelector={clickableSelector}
+            className={classNames('fullscreen-flyout', className)}
+            {...rest}
+        >
+            {button}
+            <Overlay>
+                {header || <OverlayHeader />}
+                {content}
```

Comment:
The search functionality on small screens includes a text input  in the header that updates the content. Can the FullscreenFlyout be used in this case?

Reply:
No, not yet. I left out customizability of the header initially. I was considering adding a way to customize it to have something other than just titles.

Reply:
I added customizability to the header and we can move custom implementation to EUA